Stratis.Bitcoin/Features/Consensus/PosConsensusValidator.cs

```diff
@@ -14,17 +14,30 @@ namespace Stratis.Bitcoin.Features.Consensus
     /// TODO
     /// </summary>
     /// <remarks>
-    /// These are the criteria for a new block to be accepted as a valid POS block:
+    /// These are the criteria for a new block to be accepted as a valid POS block at version 3 of the protocol, 
+    /// which has been active since 6 August 2016 07:03:21 (Unix epoch time > 1470467000). All timestamps 
+    /// are Unix epoch timestamps with seconds precision.
     /// <list type="bullet">
-    /// <item>New block's timestamp has to be strictly greater than previous block's timestamp.</item>
-    /// <item>TODO</item>
-    /// <item>TODO</item>
-    /// <item>TODO</item>
-    /// <item>TODO</item>
-    /// <item>TODO</item>
-    /// <item>TODO</item>
-    /// <item>TODO</item>
-    /// <item>TODO</item>
+    /// <item>New block's timestamp ('BlockTime') MUST be strictly greater than previous block's timestamp.</item>
+    /// <item>Coinbase transaction's (first transaction in the block with no inputs) timestamp MUST be inside interval ['BlockTime' - 15; 'BlockTime'].</item>
+    /// <item>Coinstake transaction's (second transaction in the block with at least one input and at least 2 outputs and first output being empty) timestamp
+    /// MUST be equal to 'BlockTime' and it MUST have lower 4 bits set to 0 (i.e. be divisible by 16) - see <see cref="StakeTimestampMask"/>.</item>
+    /// <item>Block's header 'nBits' field MUST be set to the correct POS target value.</item>
+    /// <item>All transactions in the block must be final, which means their 'nLockTime' is either zero, or it is lower than current block's height 
+    /// or node's 'AdjustedTime'. 'AdjustedTime' is the synchronized time among the node and its peers.</item>
+    /// <item>Coinstake transaction MUST be signed correctly.</item>
+    /// <item>Coinstake transaction's kernel (first) input MUST not be created within last <see cref="PosConsensusOptions.StakeMinConfirmations"/> blocks,
+    /// i.e. it MUST have that many confirmation at least.</item>
+    /// <item>Coinstake transaction's kernel must meet the staking target using this formula:
+    /// <code>hash(nStakeModifier + txPrev.block.nTime + txPrev.nTime + txPrev.vout.hash + txPrev.vout.n + nTime) < bnTarget * nWeight</code>
+    /// <para>
+    /// where 'txPrev' is the coinstake's kernel transaction, 'txPrev.vout' is the kernel's output in that transaction, 
+    /// 'txPrev.vout.hash' is the hash of that transaction; 'nTime' is coinstake's transaction time; 'bnTarget' is the target as 
+    /// in 'nBits' block header; 'nWeight' is the value of the kernel's input. 
+    /// </para>
+    /// </item>
+    /// <item>Block's height MUST NOT be more than 500 blocks back - i.e. reorganizations longer than 500 are not allowed.</item>
```

Comment:
>Block's height MUST NOT be more than 500 blocks back

Is that enforced in the rules?

Reply:
It is not (in C#) - https://github.com/stratisproject/StratisBitcoinFullNode/issues/413